src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java

```diff
@@ -317,7 +317,15 @@ void initializeGitScmSource() {
             repository = new BitbucketSCMRepository(getCredentialsId(), getSshCredentialsId(),
                     underlyingRepo.getProject().getName(), underlyingRepo.getProject().getKey(), underlyingRepo.getName(),
                     underlyingRepo.getSlug(), getServerId(), fetchedRepository.getMirroringDetails().getMirrorName());
-            cloneUrl = underlyingRepo.getCloneUrl(repository.getCloneProtocol()).map(BitbucketNamedLink::getHref).orElse("");
+
+            // Get the clone URL from the mirror
+            cloneUrl = fetchedRepository.getMirroringDetails().getCloneUrl(getBitbucketSCMRepository().getCloneProtocol())
```

Comment:
nit: should we stick to `repository`or `getBitbucketSCMReposity` instead of using both?

Reply:
It's much of a muchness- I wrapped in a method call so it looks identical to how the method is called from within the SCM code to help with readability, but I'm easy either way.

Reply:
It's much of a muchness- I wrapped in a method call so it looks identical to how the method is called from within the SCM code to help with readability, but I'm easy either way.

Reply:
Leave it as is :)